### PR TITLE
Upgraded django from 4.2.29 to 5.2.15.

### PR DESCRIPTION
**Pull request by hepcat72**
## What

- [GREATS-314](https://princeton-university.atlassian.net/browse/GREATS-314)

Upgrade Django to 5.2.

## Why

Django 4.2 is EOL.

## How

I manually tested in my sandbox by updating the venv and running runserver and browsing around the site.

Details:

- Updated the common requirements and added a "top level" header to satisfy markdownlint.
- Updated relevant documentation that references django 4.2.  (I checked links to make sure they still work.)

## Tests

- CI tests pass
- Manual sandbox tests:

```
source .venv311/bin/activate
python -m pip install -U pip
python -m pip install -r requirements/dev.txt
python manage.py runserver
```
After doing the above, I browsed the site and encountered no problems.

- Manual `tb9-dev` and `tb9-pub` tests:

1. tb9-{dev,pub} were upgraded to Python 3.11, Postgres 15, and Django 5.2
2. virtual environments were rebuilt
3. Cache table was cleared (due to error log reports about Django 4.2 pickled querysets)
4. Fixed 2 bugs (ZeroDivisionError in cached functions and a column switch dropdown issue on adv. srch)
5. Ran the [verification for 1.0.2](https://princeton-university.atlassian.net/wiki/x/UgDMNg) (using tb9-dev for upload and load and tb9-pub for everything else) and it passed

## Security Concerns

- No Data handling concerns
- No Authentication/authorization changes
- Updating to Django 5.2 provides security updates through April 2028.
- No Potential attack surface increase

## Others

None

### CODE REVIEW

File: DataRepo/tests/management/commands/test_load_peak_annotations.py [hunk 1]
```diff
@@ -358,7 +358,12 @@ def test_duplicate_peak_group(self):
                 adl.headers.FORMULA: peak_group.formula,
             }
         )
-        paf = ArchiveFile(filename="peak_annotation_filename.tsv")
+        paf = ArchiveFile.objects.create(
+            filename="peak_annotation_filename.tsv",
+            checksum="TEST",
+            data_type=DataType.objects.get(code="ms_peak_annotation"),
+            data_format=DataFormat.objects.get(code="isocorr"),
+        )
```
Comment (hepcat72) [inline DataRepo/tests/management/commands/test_load_peak_annotations.py:366]: Django 5.2 is more strict about foreign keys needing to be to existing records in the database.  This test had short-cutted the creation of the `ArchiveFile` record (`paf`).  When it was passed below in `adl.get_or_create_peak_group(row, paf, crd)`, a new exception occurs complaining that the `ArchiveFile` object doesn't exist in the DB.  So creating the `ArchiveFile` record fixes it.

File: requirements/common.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-Django==4.2.27
+# Common Dependencies
```
Comment (hepcat72) [inline requirements/common.txt:1]: C. Added the comment to satisfy my local `markdownlint`er.

**Review by jcmatese — APPROVED** [review]
Makes sense, and if all the tests pass and verification pans out, I approve.

File: DataRepo/models/utilities.py [hunk 3]
```diff
@@ -326,7 +326,12 @@ def resolve_field_path(
...
         field_reps: List[str] = []
         for fld in field_or_expression.get_source_expressions():
-            if isinstance(fld, str) and fld != "":
+            if fld is None:
+                # Django 5.2 now returns None values among the expressions returned by get_source_expressions.
+                # The return types in 4.2 were: Union[str, Combinable, DeferredAttribute, Field, Q, List[str]].
+                # We can safely ignore None, as it is not a field or expression.
+                continue
+            elif isinstance(fld, str) and fld != "":
                 field_reps.append(fld)
```
Comment (ro9002) [inline DataRepo/models/utilities.py:335]: in this code, an empty string still falls into the else branch and will get passed to resolve_field_path(""). Instead it could be ignored. With something like:
```
if isinstance(fld, str):
    if fld:
        field_reps.append(fld)
    continue
```

File: DataRepo/models/utilities.py [hunk 4]
```diff
@@ -326,7 +326,12 @@ def resolve_field_path(
     elif isinstance(field_or_expression, Expression):
         field_reps: List[str] = []
         for fld in field_or_expression.get_source_expressions():
-            if isinstance(fld, str) and fld != "":
+            if fld is None:
+                # Django 5.2 now returns None values among the expressions returned by get_source_expressions.
+                # The return types in 4.2 were: Union[str, Combinable, DeferredAttribute, Field, Q, List[str]].
+                # We can safely ignore None, as it is not a field or expression.
+                continue
```
Comment (ro9002) [inline DataRepo/models/utilities.py:333]: Good. Let's add a regression test here, with an expression whose `get_source_expressions()` returns something like: [None, "valid_field"]. This way, `None` is ignored, "valid_field" is preserved, and no recursive resolution is attempted on `None`.

**Review by ro9002 — CHANGES_REQUESTED** [review]
The change around `None` in `get_source_expressions()` is in the good direction for Django 5.2. Two quick changes before merging:

1. Add a regression test where `get_source_expressions()` returns `[None, "some_field"]`, to check that `None` is ignored and the valid field path is still collected.

2. Consider handling empty strings explicitly so they are not passed into `resolve_field_path()`

File: DataRepo/models/utilities.py (hunk 4 above)
Comment (hepcat72) [reply DataRepo/models/utilities.py:333]: Change incoming

File: DataRepo/models/utilities.py (hunk 3 above)
Comment (hepcat72) [reply DataRepo/models/utilities.py:335]: Change incoming

File: DataRepo/models/utilities.py [hunk 5]
```diff
@@ -326,15 +326,20 @@ def resolve_field_path(
     elif isinstance(field_or_expression, Expression):
         field_reps: List[str] = []
         for fld in field_or_expression.get_source_expressions():
-            if isinstance(fld, str) and fld != "":
+            if fld is None or (isinstance(fld, str) and fld == ""):
```
Comment (hepcat72) [inline DataRepo/models/utilities.py:329]: C. Made the `continue` conditional match empty strings too.

File: DataRepo/models/utilities.py [hunk 6]
```diff
@@ -326,15 +326,20 @@ def resolve_field_path(
...
                 tmp_field_reps = resolve_field_path(fld, _level=_level + 1, all=all)
                 if isinstance(tmp_field_reps, str):
-                    field_reps.append(tmp_field_reps)
-                else:  # Assumes list
-                    field_reps.extend(tmp_field_reps)
-        field_reps = [f for f in field_reps if f != ""]
+                    if tmp_field_reps != "":
+                        field_reps.append(tmp_field_reps)
+                elif any(f != "" for f in tmp_field_reps):  # Assumes list
+                    field_reps.extend([f for f in tmp_field_reps if f != ""])
```
Comment (hepcat72) [inline DataRepo/models/utilities.py:342]: C. Removed the old way empty strings were filtered out (old line 337) and accounted for empty strings in the list returned by `resolve_field_path`, in order to adhere to the spirit of the new continue conditional match.

File: DataRepo/tests/models/test_utilities.py [hunk 7]
```diff
@@ -477,6 +478,14 @@ def test_resolve_field_path(self):
                 ),
             )
 
+        # Ensure resolve_field_path now handled None values, returned by get_source_expression in Django 5.2
+        class MockExpression(Expression):
```
Comment (ro9002) [inline DataRepo/tests/models/test_utilities.py:482]: docstrings?

**Review by ro9002 — CHANGES_REQUESTED** [review]
Left a comment. Thanks @hepcat72

File: DataRepo/tests/models/test_utilities.py (hunk 7 above)
Comment (hepcat72) [reply DataRepo/tests/models/test_utilities.py:482]: Change incoming

**Comment by hepcat72** [comment]
Incidentally, the systems upgrade is complete on tb9.  I confirmed that the pickled querysets in the cache table **do** need to be cleared after the Django upgrade.  I tried a download from the advanced search before clearing the cache and this was the exception that after clearing the cache, did not happen:

```
[Tue Jun 23 14:08:53.179692 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637] /usr/local/tracebase/lib/python3.11/site-packages/django/core/cache/backends/db.py:96: RuntimeWarning: Pickled model instance's Django version 4.2.25 does not match the current version 5.2.15.
[Tue Jun 23 14:08:53.179723 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   value = pickle.loads(base64.b64decode(value.encode()))
[Tue Jun 23 14:08:53.179811 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637] /usr/local/tracebase/lib/python3.11/site-packages/django/core/cache/backends/db.py:96: RuntimeWarning: Pickled queryset instance's Django version 4.2.25 does not match the current version 5.2.15.
[Tue Jun 23 14:08:53.179815 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   value = pickle.loads(base64.b64decode(value.encode()))
[Tue Jun 23 14:08:53.180260 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637] mod_wsgi (pid=4176923): Exception occurred processing WSGI script '/var/www/tracebase/TraceBase/wsgi.py'.
[Tue Jun 23 14:08:53.181511 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637] Traceback (most recent call last):
[Tue Jun 23 14:08:53.181997 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/var/www/tracebase/DataRepo/views/search/download.py", line 182, in tsv_template_iterator
[Tue Jun 23 14:08:53.182009 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     yield rowtmplt.render({"qry": qry, "row": row})
[Tue Jun 23 14:08:53.182012 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182020 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/backends/django.py", line 107, in render
[Tue Jun 23 14:08:53.182023 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self.template.render(context)
[Tue Jun 23 14:08:53.182026 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182031 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 172, in render
[Tue Jun 23 14:08:53.182034 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self._render(context)
[Tue Jun 23 14:08:53.182036 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182042 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 164, in _render
[Tue Jun 23 14:08:53.182045 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self.nodelist.render(context)
[Tue Jun 23 14:08:53.182047 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182052 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 1018, in render
[Tue Jun 23 14:08:53.182055 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return SafeString("".join([node.render_annotated(context) for node in self]))
[Tue Jun 23 14:08:53.182058 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182063 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 1018, in <listcomp>
[Tue Jun 23 14:08:53.182065 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return SafeString("".join([node.render_annotated(context) for node in self]))
[Tue Jun 23 14:08:53.182068 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182077 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 979, in render_annotated
[Tue Jun 23 14:08:53.182080 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self.render(context)
[Tue Jun 23 14:08:53.182083 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182088 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/defaulttags.py", line 327, in render
[Tue Jun 23 14:08:53.182095 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return nodelist.render(context)
[Tue Jun 23 14:08:53.182098 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182102 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 1018, in render
[Tue Jun 23 14:08:53.182105 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return SafeString("".join([node.render_annotated(context) for node in self]))
[Tue Jun 23 14:08:53.182107 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182112 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 1018, in <listcomp>
[Tue Jun 23 14:08:53.182114 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return SafeString("".join([node.render_annotated(context) for node in self]))
[Tue Jun 23 14:08:53.182117 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182121 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 979, in render_annotated
[Tue Jun 23 14:08:53.182124 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self.render(context)
[Tue Jun 23 14:08:53.182127 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182130 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/loader_tags.py", line 210, in render
[Tue Jun 23 14:08:53.182133 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return template.render(context)
[Tue Jun 23 14:08:53.182136 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182140 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 174, in render
[Tue Jun 23 14:08:53.182142 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self._render(context)
[Tue Jun 23 14:08:53.182145 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182149 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 164, in _render
[Tue Jun 23 14:08:53.182151 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self.nodelist.render(context)
[Tue Jun 23 14:08:53.182154 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182158 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 1018, in render
[Tue Jun 23 14:08:53.182161 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return SafeString("".join([node.render_annotated(context) for node in self]))
[Tue Jun 23 14:08:53.182163 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182169 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 1018, in <listcomp>
[Tue Jun 23 14:08:53.182171 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return SafeString("".join([node.render_annotated(context) for node in self]))
[Tue Jun 23 14:08:53.182174 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182178 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/base.py", line 979, in render_annotated
[Tue Jun 23 14:08:53.182181 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     return self.render(context)
[Tue Jun 23 14:08:53.182183 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]            ^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182187 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/template/defaulttags.py", line 199, in render
[Tue Jun 23 14:08:53.182190 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     len_values = len(values)
[Tue Jun 23 14:08:53.182192 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                  ^^^^^^^^^^^
[Tue Jun 23 14:08:53.182197 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/query.py", line 368, in __len__
[Tue Jun 23 14:08:53.182199 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     self._fetch_all()
[Tue Jun 23 14:08:53.182204 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/query.py", line 1954, in _fetch_all
[Tue Jun 23 14:08:53.182207 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     self._result_cache = list(self._iterable_class(self))
[Tue Jun 23 14:08:53.182209 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182213 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/query.py", line 93, in __iter__
[Tue Jun 23 14:08:53.182216 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     results = compiler.execute_sql(
[Tue Jun 23 14:08:53.182218 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]               ^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182222 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1610, in execute_sql
[Tue Jun 23 14:08:53.182225 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     sql, params = self.as_sql()
[Tue Jun 23 14:08:53.182227 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                   ^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182232 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 791, in as_sql
[Tue Jun 23 14:08:53.182234 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     from_, f_params = self.get_from_clause()
[Tue Jun 23 14:08:53.182237 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                       ^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182241 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1158, in get_from_clause
[Tue Jun 23 14:08:53.182245 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     clause_sql, clause_params = self.compile(from_clause)
[Tue Jun 23 14:08:53.182248 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182252 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 577, in compile
[Tue Jun 23 14:08:53.182255 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     sql, params = node.as_sql(self, self.connection)
[Tue Jun 23 14:08:53.182257 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182261 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]   File "/usr/local/tracebase/lib/python3.11/site-packages/django/db/models/sql/datastructures.py", line 101, in as_sql
[Tue Jun 23 14:08:53.182264 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]     join_fields = self.join_fields or self.join_cols
[Tue Jun 23 14:08:53.182266 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637]                   ^^^^^^^^^^^^^^^^
[Tue Jun 23 14:08:53.182278 2026] [wsgi:error] [pid 4176923:tid 4177026] [remote 128.112.116.220:58637] AttributeError: 'Join' object has no attribute 'join_fields'
```

**Review by ro9002 — APPROVED** [review]
No further comments.
